Sources/Core/Microsoft.StreamProcessing/Aggregates/SlidingMaxAggregate.cs

```diff
@@ -44,51 +52,90 @@ private MinMaxState<T> Accumulate(MinMaxState<T> state, long timestamp, T input)
             }
             else
             {
-                if (state.currentTimestamp != InvalidSyncTime)
-                    state.savedValues.Add(state.currentValue);
+                // save only if new input is smaller, otherwise overwrite current with new
```

Comment:
>save only if new input is smaller, otherwise overwrite current with new [](start = 19, length = 71)

Maybe we should make it clear that currentValue/currentTimestamp is expiring and being overwritten